x/slashing/keeper.go

```diff
@@ -65,6 +65,7 @@ func (k Keeper) handleDoubleSign(ctx sdk.Context, pubkey crypto.PubKey, infracti
 }
 
 // handle a validator signature, must be called once per validator per block
+// nolint gocyclo
 func (k Keeper) handleValidatorSignature(ctx sdk.Context, pubkey crypto.PubKey, power int64, signed bool) {
```

Comment:
We're planning to actually *not* send the pubkey in BeginBlock, only the address (https://github.com/tendermint/tendermint/issues/1712). Is that going to be a problem? It means we need to do the lookup by validator address!

Reply:
We'll need to change the staking store then, at the moment the secondary index is by raw public key bytes.

Reply:
cc @rigelrozanski but AFAIK this should be fine